dll/cpl/sysdm/environment.c

```diff
@@ -45,97 +145,423 @@ GetSelectedListViewItem(HWND hwndListView)
     return -1;
 }
 
+static VOID
+AddValuesToListBox(HWND hwndDlg,
+                   PDIALOG_DATA DlgData)
+{
+    TCHAR szTemp[wcslen(DlgData->VarData->lpRawValue) + 1];
+    LPTSTR lpToken;
+
+    StringCchCopy(szTemp, _countof(szTemp), DlgData->VarData->lpRawValue);
+    lpToken = wcstok(szTemp, _T(";"));
+
+    while (lpToken != NULL)
+    {
+        SendDlgItemMessage(hwndDlg, IDC_LIST_VARIABLE_VALUE, LB_ADDSTRING, 0, (LPARAM)lpToken);
+        lpToken = wcstok(NULL, _T(";"));
+    }
+    SendDlgItemMessage(hwndDlg, IDC_LIST_VARIABLE_VALUE, LB_SETCURSEL, 0, 0);
+    DlgData->dwSelectedValueIndex = 0;
+}
+
+static VOID
+BrowseRequiredFile(HWND hwndDlg)
+{
+    OPENFILENAME ofn;
+    TCHAR szFile[MAX_PATH] = _T("");
+
+    ZeroMemory(&ofn, sizeof(ofn));
+
+    ofn.lStructSize = sizeof(ofn);
+    ofn.hwndOwner = hwndDlg;
+    ofn.lpstrFilter = _T("All Files (*.*)\0*.*\0");
+    ofn.lpstrFile = szFile;
+    ofn.nMaxFile = MAX_PATH;
+    ofn.Flags = OFN_EXPLORER | OFN_FILEMUSTEXIST | OFN_HIDEREADONLY;
+
+    if(GetOpenFileName(&ofn))
```

Comment:
```suggestion
    if (GetOpenFileName(&ofn))
```

Reply:
Also the usage of this function will require a link to `comdlg32` .